fai-rag-app/fai-backend/fai_backend/mail/client.py

```diff
@@ -7,48 +7,55 @@
 from fai_backend.config import settings
 from fai_backend.logger.console import console
 from fai_backend.mail.schemas import EmailPayload
+from fai_backend.settings.service import SettingsServiceFactory, SettingKey
 
 
 class MailClient(ABC):
     @abstractmethod
-    def send_mail(self, payload: EmailPayload) -> bool:
+    async def send_mail(self, payload: EmailPayload) -> bool:
         raise NotImplementedError
 
 
-def create_mail_client() -> MailClient:
-    class ConsoleMailClient(MailClient):
-        def __init__(self, send_handler: Callable[[EmailPayload], bool] = None):
-            self.send_handler = send_handler if send_handler else lambda _payload: True
+class ConsoleMailClient(MailClient):
+    def __init__(self, send_handler: Callable[[EmailPayload], bool] = None):
+        self.send_handler = send_handler if send_handler else lambda _payload: True
 
-        def send_mail(self, payload: EmailPayload) -> bool:
-            console.log(payload)
-            return self.send_handler(payload)
+    async def send_mail(self, payload: EmailPayload) -> bool:
+        console.log(payload)
+        return self.send_handler(payload)
 
-    class BrevoPayload(EmailPayload):
-        body: str = Field(..., serialization_alias='htmlContent')
 
-    class BrevoMailClient(MailClient):
-        def send_mail(self, payload: EmailPayload) -> bool:
-            payload = BrevoPayload.model_validate(payload.model_dump())
-            url = settings.BREVO_API_URL
-            headers = {
-                'accept': 'application/json',
-                'api-key': settings.BREVO_API_KEY.get_secret_value(),
-                'content-type': 'application/json',
-            }
+class BrevoPayload(EmailPayload):
+    body: str = Field(..., serialization_alias='htmlContent')
 
-            response = requests.post(
-                url,
-                json=payload.model_dump(by_alias=True, exclude_none=True),
-                headers=headers,
-            )
 
-            return response.status_code == 201
+class BrevoMailClient(MailClient):
 
-    return {
-        'console': ConsoleMailClient,
-        'brevo': BrevoMailClient,
-    }[settings.MAIL_CLIENT]()
+    async def send_mail(self, payload: EmailPayload) -> bool:
+        payload = BrevoPayload.model_validate(payload.model_dump())
+        settings_service = SettingsServiceFactory().get_service()
+        url = await settings_service.get_value(SettingKey.BREVO_API_URL)
+        key = await settings_service.get_value(SettingKey.BREVO_API_KEY)
```

Comment:
`send_mail` raises `requests.exceptions.InvalidHeader: Header part (SecretStr('**********')) from ('api-key', SecretStr('**********')) must be of type str or bytes, not <class 'pydantic.types.SecretStr'>`

Might be cause by my test environment. There is a value set for `FIXED_PIN` in my `.env`.

Reply:
This probably also applies to `OPENAI_API_KEY` and any other `SecretStr` env var.